bay/docker/hosts.py

```diff
@@ -170,3 +171,23 @@ def build_host_ip(self):
         build, which is the gateway of the default bridge network.
         """
         return self.client.inspect_network("bridge")['IPAM']['Config'][0]['Gateway']
+
+    @cached_property
+    def is_docker_for_mac(self):
+        """
+        Works out if the current install is docker for mac or docker-machine
+        based.
+        """
+        if sys.platform != "darwin":
+            return False
+        if self.url_scheme != "unix":
+            return False
+        return True
+
+    @cached_property
+    def supports_cached_volumes(self):
+        """
+        If the host supports passing the "cached" flag to volumes to speed them
+        up (implemented in Docker for Mac)
+        """
+        return self.is_docker_for_mac
```

Comment:
There doesn't appear to be an API version change I can rely on, which is a shame, only an apparent version number. I guess we could try parsing that and hoping it's not going to change in weird ways? The version I get reported is `17.05.0-ce`.